allen-cell-animated/cell-feature-explorer
# feature: Load CSV files

Problem
=======
Closes #211, "prototype direct spreadsheet import."

Adds the ability to drag and drop CSVs onto the landing page and have them be opened as a dataset. Volume viewer data and thumbnails are read from specified columns (`thumbnailPath`, `volumeviewerPath`) while all other columns are interpreted as features and can be grouped or plotted against.

Solution
========
- Changed `imageDataset` to be part of the redux-managed state, instead of an immutable global dependency.
- Fix loading of thumbnails and volumes so they do not need to be relative paths.
- Added the new `CsvRequest` image dataset type.
  - On construction, the class will parse the provided CSV for its feature data, and determine whether each feature is discrete or numeric.
  - If no discrete feature is found, `CsvRequest` auto-generates a discrete feature that bins the data for grouping.
- Added the `CsvUploadButton` component, which handles logic for initializing the CSV dataset and replacing it in redux state.
  - Added related state/reducers/actions

Changes that still need to be made:
- Page will load indefinitely if refreshed-- some sort of flag indicating that a CSV is loaded should be saved to the URL.
- Add error handling for badly-formatted CSV data

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Download this example CSV: [simple_test_dataset.csv](https://github.com/user-attachments/files/17593452/simple_test_dataset.csv)
2. Clone and run the branch.
3. Open the landing page and scroll down to the CSV import area. Drag and drop (or upload) the example CSV. 
4. The dataset should load and be interactive.


Screenshots (optional):
-----------------------
Show-n-tell images/animations here

Keyfiles (delete if not relevant):
-----------------------
1. main file/entry point
5. other important file

Thanks for contributing!
